nixos/modules/services/search/tika.nix

```diff
@@ -0,0 +1,74 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.tika;
+  inherit (lib)
+    mkIf
+    mkEnableOption
+    mkOption
+    mkPackageOption
+    getExe'
+    types
+    ;
+in
+{
+  meta.maintainers = [ lib.maintainers.drupol ];
+
+  options = {
+    services.tika = {
+      enable = mkEnableOption "Apache Tika server";
+      package = mkPackageOption pkgs "tika" { };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        example = "0.0.0.0";
+        description = ''
+          The Apache Tika bind address.
+        '';
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 8000;
+        description = ''
+          The Apache Tike port to listen on
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to open the firewall for Apache Tika.
+          This adds `services.tika.port` to `networking.firewall.allowedTCPPorts`.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.tika = {
+      description = "Apache Tika Server";
+
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+
+        ExecStart = "${getExe' cfg.package "tika-server"} --host ${cfg.listenAddress} --port ${toString cfg.port}";
```

Comment:
`tika-server` is already in `meta.mainProgram`, so why not use `${getExe cfg.package}` instead ?

Reply:
Fixed. Note: Tika ships 2 binaries, `tika-server` is set as default. If by any chance, that is modified, the service will be broken now.

Reply:
yes, but the test should catch this I think

Reply:
Oh yes definitely.

Reply:
anyway, I fixed it already.